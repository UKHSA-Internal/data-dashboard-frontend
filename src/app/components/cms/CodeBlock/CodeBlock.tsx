import clsx from 'clsx'
import { Highlight } from 'prism-react-renderer'
import Prism from 'prismjs'

import { theme } from './themes/govuk.theme'

interface CodeBlockProps {
  children: string
  language: string
  className?: string
}

;(typeof global !== 'undefined' ? global : window).Prism = Prism
require('prismjs/components/prism-json')

export const CodeBlock = ({ children, language, className }: CodeBlockProps) => {
  const lang = language.toLowerCase()

  return (
    <Highlight prism={Prism} theme={theme} code={children} language={lang}>
      {({ style, tokens, getLineProps, getTokenProps }) => (
        <pre
          style={style}
          // eslint-disable-next-line tailwindcss/no-custom-classname
          className={clsx(
            className,
            `language-${lang} govuk-!-padding-4 govuk-!-margin-bottom-6 print:text-shadow-none overflow-auto`,
            `whitespace-pre-wrap break-words font-[var(--ukhsa-font-code)] text-[var(--colour-code-text)] text-shadow-[0_1px_white]`
          )}
        >
          {tokens.map((line, i) => (
            <div key={i} {...getLineProps({ line })}>
              {line.map((token, key) => (
                <span key={key} {...getTokenProps({ token })} />
              ))}
            </div>
          ))}
        </pre>
      )}
    </Highlight>
  )
}
