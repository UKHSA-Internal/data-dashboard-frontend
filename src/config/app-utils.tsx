/**
 * Util function to get the revalidate value for NextJs getStaticProps
 * Passing 'false' will disable revalidation altogether and return
 * a stale version of the initial built page to all users.
 */
export const getStaticPropsRevalidateValue = () => {
  if (process.env.NEXT_REVALIDATE_TIME === 'false') {
    return false
  }

  return Number(process.env.NEXT_REVALIDATE_TIME)
}
