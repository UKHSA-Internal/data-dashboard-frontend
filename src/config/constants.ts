/**
 * The NextJS Api Route path for chart download
 */
export const chartExportApiRoutePath = '/api/download/chart'

/**
 * The NextJS Api Route path for bulk download
 */
export const bulkExportApiRoutePath = '/api/download/bulk'

/**
 * The default file format in which bulk downloaded charts are downloaded as
 */
export const bulkExportFormat: 'csv' | 'json' = 'csv'

/**
 * The maximum columns per table in the tabular data view
 */
export const chartTableMaxColumns = {
  narrow: 2,
  wide: 3,
} as const

/**
 * Chart size variants
 */
export const chartSizes = {
  narrow: {
    width: 515,
    height: 260,
  },
  wide: {
    width: 1100,
    height: 260,
  },
} as const

/**
 * The default file format in which charts are displayed as
 */
export const chartFormat = 'svg'
