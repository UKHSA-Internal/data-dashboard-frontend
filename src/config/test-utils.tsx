/* eslint-disable import/export,no-restricted-imports */
import React, { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { I18nextProvider } from 'react-i18next'

import i18n from '../../config/i18n/i18nForTests'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
