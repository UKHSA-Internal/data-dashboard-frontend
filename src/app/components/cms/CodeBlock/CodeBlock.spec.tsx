import { render } from '@/config/test-utils'

import { CodeBlock } from './CodeBlock'

const mockCodeSnippet = `<body><h1>hello world</h1></body>`

test('Displays the code block', () => {
  const { container } = render(<CodeBlock language="html">{mockCodeSnippet}</CodeBlock>)

  expect(container.firstChild).toHaveClass('language-html')
  expect(container).toMatchSnapshot()
})

test('Custom class name', () => {
  const { container } = render(
    <CodeBlock language="javascript" className="govuk-!-margin-4">
      {mockCodeSnippet}
    </CodeBlock>
  )

  expect(container.firstChild).toHaveClass('language-javascript')
  expect(container.firstChild).toHaveClass('govuk-!-margin-4')
})
