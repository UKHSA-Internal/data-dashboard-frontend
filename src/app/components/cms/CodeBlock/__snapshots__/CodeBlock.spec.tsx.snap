// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays the code block 1`] = `
<div>
  <pre
    class="language-html govuk-!-padding-4 govuk-!-margin-bottom-6 print:text-shadow-none overflow-auto whitespace-pre-wrap break-all font-[var(--ukhsa-font-code)] text-[var(--colour-code-text)] text-shadow-[0_1px_white]"
  >
    <div
      class="token-line"
    >
      <span
        class="token tag punctuation"
      >
        &lt;
      </span>
      <span
        class="token tag"
      >
        body
      </span>
      <span
        class="token tag punctuation"
      >
        &gt;
      </span>
      <span
        class="token tag punctuation"
      >
        &lt;
      </span>
      <span
        class="token tag"
      >
        h1
      </span>
      <span
        class="token tag punctuation"
      >
        &gt;
      </span>
      <span
        class="token plain"
      >
        hello world
      </span>
      <span
        class="token tag punctuation"
      >
        &lt;/
      </span>
      <span
        class="token tag"
      >
        h1
      </span>
      <span
        class="token tag punctuation"
      >
        &gt;
      </span>
      <span
        class="token tag punctuation"
      >
        &lt;/
      </span>
      <span
        class="token tag"
      >
        body
      </span>
      <span
        class="token tag punctuation"
      >
        &gt;
      </span>
    </div>
  </pre>
</div>
`;
